bot1.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+#include <string>
+#include <queue>
+#include <random>
+using std::string, std::cout, std::queue, std::to_string, std::pair;
+
+class Bot1{
+    private:
+    char statek = '#';
+    char puste_pole = ' ';
+    bool strzelic[10][10];
+    bool trafiony[10][10];
+    queue<pair<int, int>> kolejka_strzalow;
+    pair<int, int> kierunki[4] = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
+    pair<int, int> kierunki8[8] = {{1, 1}, {1, 0}, {1, -1}, {0, 1}, {0, -1}, {-1, 1}, {-1, 0}, {-1, -1}};
+    bool poza_plansza(int y, int x){
+        if(y >= 10 || y < 0 || x >= 10 || x < 0){
+            return true;
+        }
+        return false;
+    }
+    string wspolrzedneNaString(int kolumna, int wiersz){
+        return char(kolumna) + to_string(wiersz);
+    }
+    void decyzja(int kolumna, int wiersz){ //funkcja decydujaca co robic na podstawie czy bot trafil czy nie
+        // int werdykt = zapytaj_o_trafienie(kolumna, wiersz);
+        bool trafienie = false; 
+        bool zatopienie = false;
+        // if(werdykt == 1){
+        //     trafienie = true;
+        // }
+        // else if(werdykt == 2){
+        //     zatopienie = true;
+        // }
+        
+        // zalozmy ze mamy czesc takiej planszy:
+        // ...#...
+        // ...#...
+        // ...#...
+        // niech kolumna i wiersz wskazuja na hash srodkowy i zalozmy ze w hash najwyzej juz trafilismy w jakims strzale wczesniej
+        // sprawdzmy wiec najpierw czy dookola [kolumna-65][wiersz] jest jakis punkt ktory byl trafiony
+        // jesli byl trafiony (jest to u nas punkt nad nami), to do kolejki chcemy dodac tylko punkt pod nami, bo punkt nad nami jest juz trafiony, a punkty na lewo i prawo na pewno nie beda miec statku, bo mialby wtedy ksztalt L 
+        if(trafienie){
```

Comment:
jaki jest sens tego if-a, skoro nigdy się nie spełni?

Reply:
// int werdykt = zapytaj_o_trafienie(kolumna, wiersz);
         bool trafienie = false; 
         bool zatopienie = false;
         // if(werdykt == 1){
         //     trafienie = true;
         // }
         // else if(werdykt == 2){
         //     zatopienie = true;
         // }
         
        
spełni się jak inni napiszą funkcję która da mi odpowiedź na zapytanie o strzał. nie wiem w jaki sposób będzie napisana ta funkcja więc na razie jest to w takiej formie. Jeśli zwróci trafienie to sobie zmienie trafienie = true, a jeśli zwróci zatopienie to zmienie zatopiony = true.